telemetry/client.go

```diff
@@ -93,6 +93,7 @@ type PeerConnFailure struct {
 }
 
 type Client struct {
+	ctx                  context.Context
```

Comment:
storing context in a struct is discouraged because it hides the flow of context propagation, making the code harder to understand and maintain. 
context are meant to be passed explicitly, reflecting their ephemeral and request-scoped nature. (This is described in https://pkg.go.dev/context)

I'd say that instead of having the context being part of the struct, the `processAndPushTelemetry` should receive a context parameter and use that for the request. That means that `PushReceivedMessages`, `PushSentEnvelope`, `PushSentEnvelope`,  `PushErrorSendingEnvelope`,  `PushPeerCount` and `PushPeerConnFailures` should be modified to receive a context as well

Reply:
Wow, never saw this rule. TIL, thank you! ❤️ 

I'll do this for sure. 
Though there's we'll still pass `Messenger.ctx` everywhere 😄

Reply:
@richard-ramos ca40216ed6d811e767ad345ae99b5fcffc9b5e74

Reply:
:rocket: